src/kestrel_datasource_stixshifter/worker/transmitter.py

```diff
@@ -159,12 +167,20 @@ def retrieve_data(self):
                     if "metadata" in result_batch:
                         metadata = result_batch["metadata"]
 
+                    if self.limit:
+                        if result_retrieval_offset >= self.limit:
+                            has_remaining_results = False
+                        else:
+                            batch_size = self.limit - result_retrieval_offset
+                            if batch_size > self.retrieval_batch_size:
+                                batch_size = self.retrieval_batch_size
                 else:
                     has_remaining_results = False
 
                 is_retry_cycle = False
 
             else:
+                has_remaining_results = False
```

Comment:
Thanks for catching this logic bomb!

However, this statement should be put in the second subcase to enable retry (the first subcase) as added in the latest develop: https://github.com/opencybersecurityalliance/kestrel-lang/blob/develop/src/kestrel_datasource_stixshifter/worker/transmitter.py#L210

Reply:
Fixed in commit https://github.com/opencybersecurityalliance/kestrel-lang/pull/365/commits/e58be87e7557a852f09d08a294f68b711fd91c19